Mozillians_Repositioning.md

```diff
@@ -0,0 +1,233 @@
+  + [Mozillians.org](#mozilliansorg)
+  + [Search Service](#search-service)
+  + [Profile Publisher](#profile-publisher)
+  + [Orgchart Service](#orgchart-service)
+* [Integrity Checker](#integrity-checker)
+* [CIS requirements](#cis-requirements)
+  + [Profile Updates from Mozillians](#profile-updates-from-mozillians)
+  + [Profile Updates from external Sources](#profile-updates-from-external-sources)
+
+
+## Mozillians.org Services
+
+### Technologies (Common)
+
+#### GraphQL
+
+We stated a discussion around whether we want to test [GraphQL](https://graphql.org/) as our API
+query language. Could be a great fit and we plan to give it a shot.
+
+#### Node, Python, Rust
+
+We want to use whatever technology fits our needs. If it makes sense to write a small service in
+in Node or Rust we encourage it.
+
+
+### New Front-End
+
+A new front-end to render the new pages:
+
+- profile view
+- (edit profile section)
+  TBD if we do this _inline_ or as separate page
+- search result page
+- org chart
+
+#### Technologies
+
+- We plan to implement the new front-end using [Vue](https://vuejs.org).
+- We use components throughout across all pages
+- We plan to use global CSS instead of using components because we embrace the
+  [cascade](https://developer.mozilla.org/en-US/docs/Web/CSS/Cascade)
+- Server side rendering is put on hold for now
+
+#### Questions
+
+Will we support IE11?
+
+#### Soft Goals
+
+- Encourage community engagement by having a simple lightweight dev setup
+  - Ideally we have mocks for all APIs
+- Have a non-JS fallback.
+  - Either server side rendering or
+  - Super basic html fallback
+
+
+### Mozillians.org
+
+Is still in charge of:
+
+- Authentication
+- Authorization
+- Spam cron job
+- Group managment
+  - Membership will live in profile v2 data though
+- Business logic (vouching, access groups, …)
+
+New things:
+- API for the new front-end
+  - GraphQL?!
+- Integration off all other new services
+
+#### Views
+
+This will outline all changes to views. Which parts come from Django and which parts come from the
+new front-end.
+
+#### Group Management
+
+How we'll handle groups in phase 1.
+
+#### New APIs
+
+- single profile view
+- profile (section) edit
+- search
+- search auto-complete / suggestions
+- orgchart
+
+#### Scenarios
+
+##### Viewing a Profile
+
+- user visits https://mozillians.org/en-US/u/fiji/
+- hits Django
+- gets view with new front-end
+- new front-end requests profile data from API (Django)
+- retrieving data from [search service](#search-service) and ([orgchart service](#orgchart-service))
+- responds according to permission level
+- new front-end renders profile view
+
+##### Editing a Profile
+
+- user is on the edit profile view
+- user clicks on ✎ in a section of their profile
+- hits Django
+- gets view with new front-end
+- new front-end requests edit profile section data from API (Django)
+- new front-end renders edit profile section view
+- user changes value
+- user clicks on save button
```

Comment:
nit: decide between auto-save and save buttons =)
also, personally, i find it much nicer to use when its wysiwyg in either case (ie the edit view is the "view" view, instead of separate views)

Reply:
I had a conversation with Thomas about that. I agree with you, but Thomas seems to for more of a layover / separate page.